### PR TITLE
Add captcha to demo.html page

### PR DESCRIPTION
**Pull request by Gommorach**
- Adds captcha element to demo.html page
- Contains js file that can be deployed to aws lambda

I'm not sure if the lambda code should be moved to the ally-deploy project? For quicker development I added it here instead.

This needs manual deployment. I've added it in the ally-performance environment to test
- https://console.aws.amazon.com/apigateway/home?region=us-east-1#/apis/2m0qbgl101/resources/ij1bsk0ar6
- https://console.aws.amazon.com/lambda/home?region=us-east-1#/functions/captchaValidator?tab=graph
More notes in lambda/readme.md

This should be working when running ally.ac locally.

### CODE REVIEW

File: assets/js/ally.js [hunk 1]
```diff
@@ -174,24 +174,26 @@
...
                     $.ajax({
-                        'url': 'https://s2376.t.eloqua.com/e/f2',
+                        'url': 'https://2m0qbgl101.execute-api.us-east-1.amazonaws.com/dev',
                         'method': 'POST',
-                        'data': data,
-                        'success': requestDemoSubmitted,
-                        'error': function(err) {
+                        'data': JSON.stringify(formData),
+                        'contentType': "application/json; charset=utf-8",
+                        'success': (response) => {
```
Comment (simong) [inline assets/js/ally.js]:
```
'success': requestDemoSubmitted, 
```

File: captcha/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+\.idea/
```
Comment (simong) [inline captcha/.gitignore]: Is this .gitignore necessary? Can't we ignore all of `.idea`?

File: captcha/index.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        body
+    };
+    return fetch(url, options);
+};
+
+const generateResponse = (statusCode, body) => {
+    return {
+        statusCode,
+        headers: {
+            //"Access-Control-Allow-Origin": "ally.ac"
```
Comment (simong) [inline captcha/index.js]: Does this need to be uncommented?

File: captcha/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Demo request captcha validator
+Checks the form captcha on '/demo.html'. 
+It only submits to form to Eloqua if the captcha is valid.
+
+
+reCAPTCHA v2 is used. Security preferences (easiest for users - Most secure) and other settings can be configured in the Captcha admin panel:
+https://www.google.com/recaptcha/admin/site/346536634
```
Comment (simong) [inline captcha/readme.md:7]: Whose account is this recaptcha associated with? Are the creds in lastpass?

File: assets/js/ally.js [hunk 5]
```diff
@@ -174,24 +174,26 @@
...
-                        data = data.replace('StateorProvince=', '');
+                        formData.StateorProvince = '';
                     }
                     if (!$('#form-optin').prop('checked')) {
-                        data = data + '&OptIn=0';
+                        formData.OptIn = 0;
                     }
                     $.ajax({
-                        'url': 'https://s2376.t.eloqua.com/e/f2',
+                        'url': 'https://2m0qbgl101.execute-api.us-east-1.amazonaws.com/dev',
```
Comment (simong) [inline assets/js/ally.js:184]: Which AWS account is this running under? Is it ally-prod?

**Review by simong — CHANGES_REQUESTED** [review]
Looks good, just couple of questions.

File: assets/js/ally.js (hunk 1 above)
Comment (Gommorach) [reply assets/js/ally.js]: 👍

File: captcha/.gitignore (hunk 2 above)
Comment (Gommorach) [reply captcha/.gitignore]: 👍

File: captcha/index.js (hunk 3 above)
Comment (Gommorach) [reply captcha/index.js]: yep, `*` was for local development.

File: captcha/readme.md (hunk 4 above)
Comment (Gommorach) [reply captcha/readme.md:7]: It's a custom gmail account - hellingerlaurensally@gmail.com
I currently don't have permissions to add items to a shared folder in lastpass.

File: assets/js/ally.js (hunk 5 above)
Comment (Gommorach) [reply assets/js/ally.js:184]: It's ally-dev. Should I change it over to prod?

**Pull request merged**